compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/util/TypeDefBuilderHelper.java

```diff
@@ -215,4 +218,29 @@ public static BLangClassDefinition createClassDef(Location pos, BObjectTypeSymbo
 
         return classDefNode;
     }
+
+    public static BLangErrorType createBLangErrorType(Location pos, BType detailType, String name) {
+        BLangErrorType errorType = (BLangErrorType) TreeBuilder.createErrorTypeNode();
+        BLangUserDefinedType userDefinedTypeNode = (BLangUserDefinedType) TreeBuilder.createUserDefinedTypeNode();
+        userDefinedTypeNode.pos = pos;
+        userDefinedTypeNode.pkgAlias = (BLangIdentifier) TreeBuilder.createIdentifierNode();
+        userDefinedTypeNode.typeName = createIdentifier(pos, name);
+        userDefinedTypeNode.type = detailType;
+        errorType.detailType = userDefinedTypeNode;
+
+        return errorType;
+    }
+
+    private static BLangIdentifier createIdentifier(Location pos, String value) {
```

Comment:
can we use this https://github.com/ballerina-platform/ballerina-lang/blob/1ce30c63cabc74ed9cd293d79f83f5952d2b3d95/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/ASTBuilderUtil.java#L704

Reply:
Fixed